apps/condo/domains/resident/utils/serverSchema/findOrganizationsByAddress.js

```diff
@@ -11,112 +10,206 @@ const {
-            checkAccountNumberUrl: remoteInteractions[context.integration],
-        }]
-    }))
+    }
+
+    if (!receipts.length) receipts = null
+
+    const meters = await getOrganizationMeters(organization, addressKey, { accountNumber }, properties)
+
+    return {
+        id: organization.id,
+        name: organization.name,
+        tin: organization.tin,
+        type: organization.type,
+        receipts,
+        meters,
+    }
 }
 
-async function getOrganizationsWithMeters (organizationIds, query = {}) {
-    const organizationsWithMeters = await Promise.all(organizationIds.map(async id => {
-        const [meter] = await itemsQuery('Meter', {
-            where: { organization: { id }, deletedAt: null, ...query },
-            first: 1,
+async function getOrganizationReceipts (billingContext, addressKey, query = {}) {
+    if (!billingContext) return []
+    const billingAccounts = await find('BillingAccount', {
+        context: { id: billingContext.id },
+        property: { addressKey, deletedAt: null },
+        deletedAt: null,
+        ...query,
+    })
+    let receipts = []
+
+    if (billingAccounts.length && get(billingContext, 'lastReport.period')) {
+        receipts = await find('BillingReceipt', {
+            context: { id: billingContext.id },
+            property: { addressKey, deletedAt: null },
+            deletedAt: null,
+            period: billingContext.lastReport.period,
+            account: { id_in: billingAccounts.map(({ id }) => id) },
         })
-        return meter ? id : null
-    }))
-    return new Set(organizationsWithMeters.filter(Boolean))
+
+        const billingAccountsNumbersIndex = Object.fromEntries(
+            billingAccounts.map(account => [account.id, account.number])
+        )
+        receipts = receipts.map(receipt => ({
+            category: receipt.category,
+            balance: receipt.toPay,
+            accountNumber: billingAccountsNumbersIndex[receipt.account],
+            routingNumber: receipt.recipient.bic,
+            bankAccount: receipt.recipient.bankAccount,
+            address: get(receipt, 'raw.address', null),
+        }))
+    }
+
+    return receipts
+}
+
+async function getOrganizationMeters (organization, addressKey, query = {}, properties) {
+    let meters = await find('Meter', {
+        organization: { id: organization.id },
+        deletedAt: null,
+        property: { addressKey, deletedAt: null },
+        ...query,
+    })
+
+    const property = properties.find(p => p.addressKey === addressKey)
+    const address = property ? property.address : null
+
+    if (meters.length){
+        const meterReadings = await find('MeterReading', {
+            meter: { id_in: meters.map(({ id }) => id) },
+            deletedAt: null,
+        })
+        const meterReadingIndex = meterReadings.reduce((acc, { meter, value1, value2, value3, value4 }) => {
+            acc[meter] = [value1, value2, value3, value4].filter(Boolean).join(',')
```

Comment:
Return all values separately. This join hack does not look like a proper solution to me